### PR TITLE
Bump Knative/K8s dependencies

### PR DESCRIPTION
**Pull request by mattmoor**
# Changes

This incorporates the breaking changes from K8s 1.18 client libs through cli.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
NONE
```

/hold

### CODE REVIEW

**Comment by mattmoor** [comment]
This depends on:
 * pipelines: https://github.com/tektoncd/pipeline/pull/3319
 * triggers: https://github.com/tektoncd/triggers/pull/784

**Comment by mattmoor** [comment]
At least the unit test failures here should be fixed by: https://github.com/tektoncd/pipeline/pull/3328

**Comment by mattmoor** [comment]
I am hoping that https://github.com/tektoncd/pipeline/pull/3348 fixes the manifestation of the new `structured-merge-diff` problems here 🤞

**Comment by mattmoor** [comment]
Alright, it does...

**Review by vdemeester — APPROVED** [review]
/cc @danielhelfand @piyush-garg @pradeepitm12 @chmouel

File: go.mod [hunk 1]
```diff
@@ -9,29 +9,29 @@ require (
...
+	github.com/google/go-cmp v0.5.2
 	github.com/hako/durafmt v0.0.0-20191009132224-3f39dc1ed9f4
 	github.com/hinshun/vt10x v0.0.0-20180809195222-d55458df857c
 	github.com/jonboulle/clockwork v0.1.1-0.20190114141812-62fb9bc030d1
 	github.com/ktr0731/go-fuzzyfinder v0.2.0
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/cobra v1.0.0
 	github.com/spf13/pflag v1.0.5
-	github.com/tektoncd/pipeline v0.16.3
+	github.com/tektoncd/pipeline v0.17.1-0.20201006183654-1710b688c023
```
Comment (piyush-garg) [inline go.mod:20]: is it fine to bump to a commit in master branch, we may need first to bump to 0.17 branch and then do the release based on 0.17.x

File: go.mod [hunk 2]
```diff
@@ -42,10 +42,16 @@ replace (
 	github.com/spf13/cobra => github.com/chmouel/cobra v0.0.0-20200107083527-379e7a80af0c
 )
 
-// Pin k8s deps to 0.17.6
+// TODO(mattmoor): DO NOT SUBMIT
```
Comment (piyush-garg) [inline go.mod:45]: WIP /hold

File: go.mod (hunk 1 above)
Comment (vdemeester) [reply go.mod:20]: I think it's fine, we don't necessary need to do a release for each pipeline release. but, to be discussed

**Comment by mattmoor** [comment]
/hold

We need to merge the upstream PRs, this is working off my fork!

File: go.mod (hunk 2 above)
Comment (mattmoor) [reply go.mod:45]: FWIW, we have a Github action that will flag `DO NOT SUBMIT` lines and add a failing check to the PR.

Happy to send y'all a PR with that if you have any interest?  @vdemeester @ImJasonH @bobcatfish @afrittoli

**Comment by mattmoor** [comment]
/hold cancel

Alright, I believe all of the dependent PRs have landed and this is RFAL.

**Review by danielhelfand — APPROVED** [review]
I'll let one more set of eyes take a look, but looks good to me.

**Comment by mattmoor** [comment]
@danielhelfand fwiw, @vdemeester had a look above before my upstream PRs landed (and I dealt with merge conflicts).

**Comment by mattmoor** [comment]
cc @n3wscott reviewed a lot of the Knative 0.18 migration PRs

**Pull request merged**